services/web/server/src/simcore_service_webserver/meta_handlers.py

```diff
@@ -0,0 +1,276 @@
+from pydantic import BaseModel
+from pydantic.fields import Field
+from pydantic.networks import HttpUrl
+
+from ._meta import api_version_prefix as VTAG
+from .meta_iterations import ProjectIteration
+from .meta_version_control import VersionControlForMetaModeling
+from .rest_utils import RESPONSE_MODEL_POLICY
+from .security_decorators import permission_required
+from .utils_aiohttp import create_url_for_function, envelope_json_response
+from .version_control_models import CheckpointID, CommitID, TagProxy
+from .version_control_tags import parse_wcopy_project_tag_name
+
+log = logging.getLogger(__name__)
+
+
+# HANDLER'S CORE IMPLEMENTATION ------------------------------------------------------------
+
+IterationTuple = Tuple[ProjectID, CommitID]
+
+
+async def list_project_iterations(
+    vc_repo: VersionControlForMetaModeling,
+    project_uuid: ProjectID,
+    commit_id: CommitID,
+    offset: int = 0,
+    limit: Optional[int] = None,
+) -> Tuple[List[IterationTuple], int]:
+
+    assert offset >= 0  # nosec
+
+    repo_id = await vc_repo.get_repo_id(project_uuid)
+    assert repo_id is not None
+
+    total_number_of_iterations = 0  # count number of iterations
+
+    # Search all subsequent commits (i.e. children) and retrieve their tags
+    # Select range on those tagged as iterations and returned their assigned wcopy id
+    # Can also check all associated project uuids and see if they exists
+    # FIXME: do all these operations in database
+    tags_per_child: List[List[TagProxy]] = await vc_repo.get_children_tags(
+        repo_id, commit_id
+    )
+
+    iterations: List[Tuple[ProjectID, CommitID]] = []
+    for n, tags in enumerate(tags_per_child):
+
+        # FIXME: the db query did not guarantee
+        # not sorted
+        # not limited
+        # not certain if tags we need
+        try:
+            iteration: Optional[ProjectIteration] = None
+            wcopy_id: Optional[ProjectID] = None
+
+            for tag in tags:
+                if pim := ProjectIteration.from_tag_name(
+                    tag.name, return_none_if_fails=True
+                ):
+                    if iteration:
+                        raise ValueError(
+                            f"This entry has more than one iteration {tag=}"
+                        )
+                    iteration = pim
+                elif pid := parse_wcopy_project_tag_name(tag.name):
+                    if wcopy_id:
+                        raise ValueError(f"This entry has more than one wcopy  {tag=}")
+                    wcopy_id = pid
+                else:
+                    log.debug("Got %s for children of %s", f"{tag=}", f"{commit_id=}")
+
+            if not wcopy_id:
+                raise ValueError("No working copy found")
+            if not iteration:
+                raise ValueError("No iteration tag found")
+
+            total_number_of_iterations = max(
+                total_number_of_iterations, iteration.total_count
+            )
```

Comment:
the meaning of the snippet or why the code is there?

OK, in the context of the for loop, this parts basically does nothing (probably missed in a refactoring) since it gets set at the end again. Will remove.
```python

total_number_of_iterations = 0 
for tags in tags_per_child:
    ...
    total_number_of_iterations = max( ... )

total_number_of_iterations = len(iterations)
```